edx_sg_block/static/js/src/edx_sga.js

```diff
@@ -130,7 +130,7 @@ function StaffGradedXBlock(runtime, element) {
                 var max_score = row.parents('#grade-info').data('max_score');
                 var score = Number(form.find('#grade-input').val());
                 event.preventDefault();
-                if (!score) {
+                if (!score && (score !== 0)) {
```

Comment:
`if (typeof score !== 'number')` ?

Reply:
the only bad case is 0, everything else is fetched in else if statements

Reply:
Sure, but if you are outputting an error that says `Grade must be a number` it makes sense to check if the grade is a number.